src/api/services.js

```diff
@@ -1,109 +1,119 @@
 const token = sessionStorage.getItem("token");
 
-exports.login =  function(data, callback){
-    api.post(addr+"/auth/login/",
-    {"content-type": "application/json"},
-    data, function(error, token){
-        //callback of the method here
-        console.log("error", error);
-         if(error){
-            console.log(error);
-            return callback(error);
-         }else{
-            return callback(null, token.auth_token);
-         }
-        });
-}
-
-exports.addChew =  function(data, callback){
-   api.post(addr+"/api/v1/chews",
-   {"content-type": "application/json",
-   "Authorization": "Token "+token
-},
-   data, function(error, response){
-       //callback of the method here
-       console.log("error", error);
-        if(error){
-           console.log(error);
-           return callback(error);
-        }else{
-           return callback(null, response);
-        }
-       });
-}
-exports.addMidwife =  function(data, callback){
-   api.post(addr+"/api/v1/midwives",
-   {"content-type": "application/json",
-   "Authorization": "Token "+token
-},
-   data, function(error, response){
-       //callback of the method here
-       console.log("error", error);
-        if(error){
-           console.log(error);
-           return callback(error);
-        }else{
-           return callback(null, response);
-        }
-       });
-}
-exports.addAmbulance =  function(data, callback){
-   api.post(addr+"/api/v1/ambulances",
-   {"content-type": "application/json",
-   "Authorization": "Token "+token
-},
-   data, function(error, response){
-       //callback of the method here
-       console.log("error", error);
-        if(error){
-           console.log(error);
-           return callback(error);
-        }else{
-           return callback(null, response);
-        }
-       });
-}
+exports.login = function(data, callback) {
+  api.post(
+    addr + "/auth/login/",
+    { "content-type": "application/json" },
+    data,
+    function(error, response) {
+      if (error) {
+        return callback(error);
+      } else {
+        let district = response.user && response.user.village && response.user.village.parish && response.user.village.parish.sub_county && response.user.village.parish.sub_county.county && response.user.village.parish.sub_county.county.district && response.user.village.parish.sub_county.county.district;
+          sessionStorage.removeItem('district');
+          sessionStorage.removeItem('username');
+          sessionStorage.removeItem('token')
+          sessionStorage.setItem('district', district.name);
+          sessionStorage.setItem('username', response.user && response.user.username);
+           sessionStorage.setItem('token', response.auth_token);
+        return callback(null, response.auth_token);
+      }
+    }
+  );
+};
+exports.addUser = function(data, callback) {
+  api.post(
+    addr + "/auth/register/",
+    { "content-type": "application/json", Authorization: "Token " + token },
```

Comment:
move this into a constant as its used alot for the different endpoints.
So just do a const OPTIONS = 
{ "content-type": "application/json", Authorization: "Token " + token }

then here instead of saying { "content-type": "application/json", Authorization: "Token " + token } just say 
add + "/auth/register/", OPTIONS, etc